### ISSUE TITLE
(Scala 3) completions are more sensitive than Scala 2's

### ISSUE THREAD

**Issue by kubukoz** [scalameta/metals#4656]
### Describe the bug

In Scala 3 completions, it seems like whatever algorithm was used for fuzzy finding in Scala 2 has been changed to a more strict one. It's limiting the amount of completions that you get, and could be considered a defect in the functionality.

To reproduce:

```scala
//> using scala "2.13.10"

object MyInterface {
  def someMethod(s: String): Unit = ()
}

object demo {
  MyInterface.<CURSOR>
}
```

If you start typing at `<CURSOR>` and write `m`, you'll see a completion for the method:

<img width="737" alt="image" src="https://user-images.githubusercontent.com/894884/202911093-ab2e56cb-f756-4fb5-a2d5-df8f73a1f0ad.png">

In Scala 3, you don't see it:

<img width="728" alt="image" src="https://user-images.githubusercontent.com/894884/202911088-32ecd0d1-9939-4f85-a0d2-2eacd0c4c848.png">

### Expected behavior

The method completion shows up.

### Operating system

macOS

### Editor/Extension

VS Code

### Version of Metals

v0.11.9

**Comment by tgodzik** [comment]
Thanks for reporting! The issue is that the compiler is trying to filter the results first, we need to change that upstream I think.

**Comment by tgodzik** [comment]
We would need to change it around here https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/interactive/Completion.scala#L136

**Comment by tgodzik** [comment]
yep, Scala 2 allowed the downstream tools to use any kind of matcher
